packages/pipeline/src/parsers/copper/index.ts

```diff
@@ -0,0 +1,274 @@
+}
+// custom fields
+export enum CopperCustomFieldType {
+    String = 'String',
+    Text = 'Text',
+    Dropdown = 'Dropdown',
+    MultiSelect = 'MultiSelect', // not in API documentation but shows up in results
+    Date = 'Date',
+    Checkbox = 'Checkbox',
+    Float = 'Float',
+    URL = 'URL',
+    Percentage = 'Percentage',
+    Currency = 'Currency',
+    Connect = 'Connect',
+}
+export interface CopperCustomFieldOptionResponse {
+    id: number;
+    name: string;
+}
+export interface CopperCustomFieldResponse {
+    id: number;
+    name: string;
+    data_type: CopperCustomFieldType;
+    options?: CopperCustomFieldOptionResponse[];
+}
+/**
+ * Parse response from Copper API /search/leads/
+ *
+ * @param leads - The array of leads returned from the API
+ * @returns Returns an array of Copper Lead entities
+ */
+export function parseLeads(leads: CopperLeadResponse[]): CopperLead[] {
+    return leads.map(lead => {
+        const entity = new CopperLead();
+        entity.id = lead.id;
+        entity.name = lead.name;
+        entity.firstName = lead.first_name;
+        entity.lastName = lead.last_name;
+        entity.middleName = lead.middle_name;
+        entity.assigneeId = lead.assignee_id;
+        entity.companyName = lead.company_name;
+        entity.customerSourceId = lead.customer_source_id;
+        entity.monetaryValue = lead.monetary_value;
+        entity.status = lead.status;
+        entity.statusId = lead.status_id;
+        entity.title = lead.title;
+
+        entity.dateCreated = lead.date_created * ONE_SECOND;
+        entity.dateModified = lead.date_modified * ONE_SECOND;
+
+        entity.email = lead.email === null ? 0 : 1;
+
+        // nullable fields
+        entity.socials = 0;
+        entity.websites = 0;
+        entity.phoneNumbers = 0;
+        try {
+            entity.socials = lead.socials.length;
+            entity.websites = lead.websites.length;
+            entity.phoneNumbers = lead.phone_numbers.length;
+        } catch (e) {
+            return entity;
+        }
+
+        return entity;
+    });
+}
+
+/**
+ * Parse response from Copper API /search/activities/
+ *
+ * @param activities - The array of activities returned from the API
+ * @returns Returns an array of Copper Activity entities
+ */
+export function parseActivities(activities: CopperActivityResponse[]): CopperActivity[] {
+    return activities.map(activity => {
+        const entity = new CopperActivity();
+        entity.id = activity.id;
+
+        entity.parentId = activity.parent.id;
+        entity.parentType = activity.parent.type;
+
+        entity.typeId = activity.type.id;
+        entity.typeCategory = activity.type.category.toString();
+        entity.typeName = activity.type.name;
+
+        entity.userId = activity.user_id;
+        entity.dateCreated = activity.date_created * ONE_SECOND;
+        entity.dateModified = activity.date_modified * ONE_SECOND;
+
+        // nullable fields
+        try {
```

Comment:
Similarly, I don't think this is the right way to handle nullable fields. It looks like if any of the fields are `undefined` it will result in all of them being `null` in the database.

Can we use [`R.path`](https://ramdajs.com/docs/#path) or [`R.pathOr`](https://ramdajs.com/docs/#pathOr) instead?